evm_arithmetization/src/cpu/cpu_stark.rs

```diff
@@ -265,7 +265,10 @@ pub(crate) fn ctl_data_byte_packing_push<F: Field>() -> Vec<Column<F>> {
 pub(crate) fn ctl_filter_byte_packing_push<F: Field>() -> Filter<F> {
     let bit_col = Column::single(COL_MAP.opcode_bits[5]);
     Filter::new(
-        vec![(Column::single(COL_MAP.op.push_prover_input), bit_col)],
+        vec![(
```

Comment:
Filter should be `COL_MAP.op.push_prover_input * COL_MAP.general.push().push_prover_input_not_kernel`.
Note that we must make sure that the general column is set to 0 in case of a `PROVER_INPUT` op. I believe this is always the case, but we can manually set it during trace generation to be extra safe.

Reply:
Ah right messed up the filter sorry.
As for the comment on `PROVER_INPUT`, I think we should instead constrain it within `control_flow`.

Reply:
Ah yeah, that's probably better.

Reply:
Wait, no the filter is correct. `push_prover_input_not_kernel` holds `push_prover_input * !is_kernel_mode`. By multiplying it with `bit_col` we do get the filter we want, i.e. the previous one multiplied by `!is_kernel_mode`.
The issue here is the missing constraint on the general column with PROVER_INPUT.

Reply:
I've addressed the fix on filtering.
As for adding this during trace generation, this is not necessary. Rows are empty initialized, and only a few specific instructions alter the general columns, which are filtered out in the constraints by the operation flag.